File: scrap-invoice-backend/models/customer.go
package models

import "time"

type Customer struct {
	ID        int       `gorm:"column:customer_id;primaryKey" json:"id"`
	Name      string    `json:"name"`
	Phone     string    `json:"phone"`
	Email     string    `gorm:"unique" json:"email"` // <-- Ini masalahnya
	Address   string    `json:"address"`
	Tier      string    `json:"tier"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

File: scrap-invoice-frontend/src/types/customer.ts
export interface Customer {
  id: number
  name: string
  phone: string
  email: string
  address: string
  tier: 'silver' | 'gold' | 'platinum'
  created_at?: string
  updated_at?: string
}

export interface CustomerForm {
  name: string
  phone: string
  email: string
  address: string
  tier: 'silver' | 'gold' | 'platinum'
}